src/Hooks/SiteEnvLookup.php

```diff
@@ -0,0 +1,206 @@
+<?php
+
+namespace Pantheon\Terminus\Hooks;
+
+use Consolidation\AnnotatedCommand\AnnotationData;
+use Pantheon\Terminus\Session\SessionAwareInterface;
+use Pantheon\Terminus\Session\SessionAwareTrait;
+use Psr\Log\LoggerAwareInterface;
+use Psr\Log\LoggerAwareTrait;
+use Robo\Contract\ConfigAwareInterface;
+use Robo\Common\ConfigAwareTrait;
+use Pantheon\Terminus\Site\SiteAwareInterface;
+use Pantheon\Terminus\Site\SiteAwareTrait;
+
+/**
+ * Class SiteEnvLookup
+ * @package Pantheon\Terminus
+ */
+class SiteEnvLookup implements ConfigAwareInterface, LoggerAwareInterface, SiteAwareInterface
+{
+    use ConfigAwareTrait;
+    use LoggerAwareTrait;
+    use SiteAwareTrait;
+
+    /**
+     * Determine the site and environment that this command should target.
+     * The Annotated Commands hook manager will call this function during
+     * the init phase of all commands. For those commands that have a
+     * parameter  'site_env' or 'site', this hook will attempt to fill in
+     * a default value for that parameter if the user did not provide it
+     * on the command line.
+     *
+     * @hook init *
+     */
+    public function siteAndEnvLookupHook($input, AnnotationData $annotationData)
+    {
+        // For commands with a $site_env parameter, inject the site and
+        // environment if it has not already been defined.
+        if ($input->hasArgument('site_env')) {
```

Comment:
Note that this implementation predicates its behavior based on the NAME of the site_env or site parameter. It would also be possible to require commands to specifically add an annotation if they wanted this behavior. However, even with the annotation we'd need to make sure that we were referencing the correct argument, so that solution, while more explicit, would also be a bit more finicky to set up.